src/common/components/fileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,165 @@
+import { useEffect, useRef, useState } from 'react';
+import { useMutation } from 'react-query';
+import { FileInput, IconCheckCircleFill, LoadingSpinner } from 'hds-react';
+import { useTranslation } from 'react-i18next';
+import { differenceBy } from 'lodash';
+import { AxiosError } from 'axios';
+import useLocale from '../../hooks/useLocale';
+import { AttachmentMetadata } from '../../types/attachment';
+import { Flex } from '@chakra-ui/react';
+import Text from '../text/Text';
+import styles from './FileUpload.module.scss';
+import { removeDuplicateAttachments } from './utils';
+
+function useDragAndDropFiles() {
+  const ref = useRef<HTMLDivElement>(null);
+  const files = useRef<File[]>([]);
+
+  useEffect(() => {
+    function dropHandler(ev: DragEvent) {
+      if (ev.dataTransfer) {
+        files.current = Array.from(ev.dataTransfer.files);
+      }
+    }
+
+    if (ref.current) {
+      ref.current.ondrop = dropHandler;
+    }
+  }, []);
+
+  return { ref, files };
+}
+
+function SuccessNotification({
+  successfulCount,
```

Comment:
There's a couple of sonar code smells (props should be read only here), should they be fixed?

Reply:
I don't know if it's necessary to fix them as eslint throws error if there is parameter reassigning.

Reply:
Fixed them anyway.